### PR TITLE
Update devbox_setup.md

### PR DESCRIPTION
**Pull request by ericmitt**
Change title and added few precision on vcpkg usage and experince 
Update version of VS
Few typo

 

# Checklist
- [x] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-c/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [x] This pull-request is submitted against the `master` branch. 
  - [ ] I have merged the latest `master` branch prior to submission and re-merged as needed after I took any feedback.
  - [ ] I have squashed my changes into one with a clear description of the change.

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

# Description of the solution

### CODE REVIEW

File: doc/devbox_setup.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ This document describes how to prepare your development environment to use the *
 
 ## Set up a Windows development environment
 
-- Install [Visual Studio 2017][visual-studio]. You can use the **Visual Studio Community** Free download if you meet the licensing requirements.  (**Visual Studio 2015** is also supported.)
+- Install [Visual Studio 2019][visual-studio]. You can use the **Visual Studio Community** Free download if you meet the licensing requirements.  (**Visual Studio 2015** and **Visual Studio 2017**  are also supported.)
```
Comment (drwill-ms) [inline doc/devbox_setup.md]: Consolidate 2 spaces to 1 between "**Visual Studio 2017** and "are also supported"

Comment (drwill-ms) [inline doc/devbox_setup.md]: Should "Free" be capitalized?

File: doc/devbox_setup.md [hunk 2]
```diff
@@ -34,9 +34,11 @@ git submodule update --init
...
+### Build sample application using vcpkg to build the SDK 
 
-The sample applications build with the help of C SDK libraries and headers published from vcpkg (a library manager). To install the C SDK libraries and headers, follow the steps to [Setup C SDK vcpkg for Windows development environment](https://github.com/Azure/azure-iot-sdk-c/blob/master/doc/setting_up_vcpkg.md#setup-c-sdk-vcpkg-for-windows-development-environment).
+The sample applications build with the help of C SDK libraries and headers built with vcpkg (a C++ package manager who facilitate building C and C++ libraries). To install the C SDK libraries and headers, follow these steps [Setup C SDK vcpkg for Windows development environment](https://github.com/Azure/azure-iot-sdk-c/blob/master/doc/setting_up_vcpkg.md#setup-c-sdk-vcpkg-for-windows-development-environment).
+
+note: vcpkg creates a directory with all the headers and .lib files on your machine, if you are using Visual Studio ( from 2015) the command 'vcpkg integrate install' let Visual Studio knows where are the headers and lib. If you're using other IDE just add the vcpkg directories to your compiler as usual.
```
Comment (drwill-ms) [inline doc/devbox_setup.md]: remove extra space in "( from 2015)"

File: doc/devbox_setup.md [hunk 3]
```diff
@@ -71,7 +74,7 @@ cd cmake
 
 > This builds x86 libraries. To build for x64 for Visual Studio 2015, modify the cmake generator argument: `cmake .. -G "Visual Studio 14 2015 Win64"` or for Visual Studio 2017, `cmake .. -G "Visual Studio 15 2017 Win64"`
```
Comment (drwill-ms) [inline doc/devbox_setup.md]: What about VS 2019?

File: doc/devbox_setup.md [hunk 4]
```diff
@@ -50,8 +52,9 @@ static const char* connectionString = "[device connection string]";
 ...and replace `[device connection string]` with a valid device connection string for a device registered with your IoT Hub. For more information, see the [samples section](#samplecode) below.
 
 Build the sample project.
+ 
 
-### Build the C SDK in Windows
+### Build the C SDK with CMake on Windows
 
 In some cases, you may want to build the SDK locally for development and testing purposes. First, take the following steps to generate project files:
```
Comment (drwill-ms) [inline doc/devbox_setup.md]: What about VS 2019 in the below commands?

Comment (ericmitt) [reply doc/devbox_setup.md]: New generator for 2019, updated

**Pull request merged**